src/integrations/stripe.ts

```diff
@@ -127,6 +139,51 @@ export async function setupPaymentMethod(
           type: servar.type,
           displayText: servar.name
         };
+
+        //
+        // Payment sequence
+        //
+        const { intent_secret: paymentIntentSecret } =
+          await client.createPayment(servar.key);
+        // No paymentIntentSecret means no payment will be done
+        if (paymentIntentSecret) {
+          const result = await stripe.confirmCardPayment(paymentIntentSecret, {
+            payment_method: paymentMethodId
+          });
+          if (result.error)
+            return {
+              errorField: servar,
+              errorMessage: result.error.message
+            };
+          else {
+            // Complete the payment (clears fields, sets paid indicator)
+            const { product_fields_to_clear: productFieldKeys } =
+              await client.paymentComplete();
+
+            // clear the selected product hidden fields
+            productFieldKeys.forEach((key: string) =>
```

Comment:
instead of doing all this logic below to manually figure out which fields to clear, could we just have the backend return updated field value mappings we can apply directly to updateFieldValues?? i think whatever business logic we can move into the backend, we should be momving